### PR TITLE
Viite 1985 update roadway points inside links specific us

### PR DESCRIPTION
**Pull request by Campanha**

### CODE REVIEW

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/NodesAndJunctionsService.scala [hunk 1]
```diff
@@ -75,5 +77,292 @@ class NodesAndJunctionsService() {
...
+      val (startNodeReversed, endNodeReversed) = {
+        (roadwayChanges.exists( ch =>
+          headLink.startAddrMValue == ch.changeInfo.target.startAddressM.get && ch.changeInfo.reversed
+        ),
+          roadwayChanges.exists( ch =>
+            lastLink.endAddrMValue == ch.changeInfo.target.endAddressM.get && ch.changeInfo.reversed
+          ))
+      }
+
+      val existingHeadNodePoint = nodePointDAO.fetchNodePointTemplate(headLink.roadwayNumber).filter(np => np.beforeAfter == After && np.addrM == headLink.startAddrMValue)
```
Comment (skosonen) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/NodesAndJunctionsService.scala]: There are compile errors:
Missing:
nodePointDAO.fetchNodePointTemplate
nodePointDAO.update

**Comment by skosonen** [comment]
There are compilation errors:
`       val existingHeadNodePoint = nodePointDAO.fetchNodePointTemplate(headLink.roadwayNumber).filter(np => np.beforeAfter == After && np.addrM == headLink.startAddrMValue)
                                                ^
       val existingLastNodePoint = nodePointDAO.fetchNodePointTemplate(lastLink.roadwayNumber).filter(np => np.beforeAfter == Before && np.addrM == lastLink.endAddrMValue)
                                                ^
           nodePointDAO.update(Seq(existingHeadNodePoint.head.copy(beforeAfter = if(!startNodeReversed) BeforeAfter.After else BeforeAfter.Before)))
                        ^
           nodePointDAO.update(Seq(existingLastNodePoint.head.copy(beforeAfter = if(!endNodeReversed) BeforeAfter.Before else BeforeAfter.After)))
                        ^
         linearLocationDAO.fetchLinearLocationByBoundingBox(boundingRectangle, roadNumberLimits)
                           ^
         linearLocationDAO.fetchLinearLocationByBoundingBox(boundingRectangle, roadNumberLimits)
                           ^
         linearLocationDAO.fetchLinearLocationByBoundingBox(boundingRectangle, roadNumberLimits)
                           ^
         linearLocationDAO.fetchLinearLocationByBoundingBox(boundingRectangle, roadNumberLimits)
                           ^
       val linearLocations = linearLocationDAO.fetchLinearLocationByBoundingBox(boundingRectangle, roadNumberLimits)
                                               ^
     val linearLocations = linearLocationDAO.fetchLinearLocationByBoundingBox(boundingRectangle, roadNumberLimits)
                                             ^
`

File: digiroad2-oracle/src/main/resources/db/migration/V0_23__drop_node_id_constraint.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+ALTER TABLE NODE_POINT DROP CONSTRAINT NODE_POINT_FK2;
```
Comment (skosonen) [inline digiroad2-oracle/src/main/resources/db/migration/V0_23__drop_node_id_constraint.sql]: Why this constraint needs to be dropped?

Comment (Campanha) [reply digiroad2-oracle/src/main/resources/db/migration/V0_23__drop_node_id_constraint.sql]: NODE_ID could not exist in node point templates

**Pull request merged**